### PR TITLE
first

### PR DESCRIPTION
**Pull request by voloshynamaryna11**

### CODE REVIEW

File: src/main/java/spring/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import spring.dao.UserDao;
+import spring.model.User;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (Vosker) [inline src/main/java/spring/dao/impl/UserDaoImpl.java]: I am not sure whether autowired is necessary here. You can remove it

File: src/main/java/spring/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.stereotype.Service;
+import spring.dao.UserDao;
+import spring.model.User;
+import spring.service.UserService;
+
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserDao userDao;
+
+    @Autowired
```
Comment (Vosker) [inline src/main/java/spring/service/impl/UserServiceImpl.java]: here as well

File: src/main/java/spring/dao/impl/UserDaoImpl.java (hunk 1 above)
Comment (voloshynamaryna11) [reply src/main/java/spring/dao/impl/UserDaoImpl.java]: yes, thanks

File: src/main/java/spring/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (voloshynamaryna11) [reply src/main/java/spring/service/impl/UserServiceImpl.java]: done

File: src/main/java/spring/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void add(User user) {
+        sessionFactory.openSession().save(user);
```
Comment (Sofasmile) [inline src/main/java/spring/dao/impl/UserDaoImpl.java]: 5.  https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_30.html

File: src/main/java/spring/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @Override
+    public List<User> listUsers() {
+        TypedQuery<User> query = sessionFactory.openSession()
```
Comment (Sofasmile) [inline src/main/java/spring/dao/impl/UserDaoImpl.java]: 5. https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_30.html

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**